dask_kubernetes/core.py

```diff
@@ -500,27 +514,58 @@ def f(to_close):
                     raise
 
     async def _cleanup_terminated_pods(self, pods):
-        terminated_phases = {"Succeeded", "Failed"}
+        terminated_phases = {PodStatus.succeeded.value, PodStatus.failed.value}
         terminated_pods = [p for p in pods if p.status.phase in terminated_phases]
         await self._delete_pods(terminated_pods)
-        return [p for p in pods if p.status.phase not in terminated_phases]
+        pods = [p for p in pods if p.status.phase not in terminated_phases]
+        return pods
 
     def scale_up(self, n, **kwargs):
-        self.scheduler.loop.add_callback(self._scale_up, n, **kwargs)
+        self._manual_scale_target = n
+        self.scheduler.loop.add_callback(self._scale_up, **kwargs)
 
-    async def _scale_up(self, n, pods=None, **kwargs):
+    async def check_status(self, status_check=PodStatus.running):
+        while True:
+            pods = await self._pods()
+            phases = {p.metadata.name: p.status.phase for p in pods}
+            result = [status_check.value == status for name, status in phases.items()]
+
+            # throttle loop
+            await asyncio.sleep(0.5)
```

Comment:
Ideally if things go smoothly the first time it would be good to not have any unnecessary slowdown.  This 500ms delay is probably better placed at the end of a loop like this, hopefully after a potential break/return.

Reply:
As a reference point, for interactive workloads (which dask-kubernetes is sometimes used for) we probably are about anything over 100ms, especially if they start to build on each other.

If we're just doing a periodic cleanup check then yeah, a few seconds or whatever is fine.  But when you hit compute you want to see workers start up more or less immediately.

Reply:
The comment above is just for general context, not for this particular sleep.